### ISSUE TITLE
Normal cut/copy operations should cut/copy lines if not selection

### ISSUE THREAD

**Issue by tmssngr** [geany/geany#1953]
I know, that there are separate commands available for cut/copy lines, but because of IntelliJ IDEA I'm used to just cut/copy the line using the same shortcut as for cut/copy selection. Hence an option to cut/copy the whole line without a selection would be perfect.

Would this be possible to implement without the need to modify Scintilla?

**Comment by elextr** [comment]
This has been discussed previously and rejected.  To summarise the thread, the fact that Geany does not lose the clipboard contents when accidentally copying instead of pasting has saved several devs a number of times, so no they won't be implementing it any time soon.

If someone provides a PR that has a preference to control it, that defaults to current behaviour, it might be accepted.

**Comment by tmssngr** [comment]
Of course, it should be optional - to not frustrate those who aren't accustomed to it. BTW, IDEA also supports a clipboard history, so with Ctrl+Shift+V you will get a dialog where you can select from the previous clipboard contents.

**Comment by elextr** [comment]
Clipboards are managed by Scintilla, probably best to ask there (though I think its been suggested before IIRC).

**Comment by codebrainz** [comment]
You could probably implement this in a plugin, they have full access to Scintilla API.

**Comment by elextr** [comment]
Or maybe one of the clipboard managers to suit your desktop, eg gpaste or klipper or klipman.
